src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/NewLineUserSettingFormattingRule.cs

```diff
@@ -313,10 +315,12 @@ private static bool IsControlBlock(SyntaxNode node)
             // new MyObject { - Object Initialization
             // new List<int> { - Collection Initialization
             // with { - Record with initializer
-            if (currentToken.Kind() == SyntaxKind.OpenBraceToken &&
-                (currentToken.Parent.Kind() == SyntaxKind.ObjectInitializerExpression ||
-                currentToken.Parent.Kind() == SyntaxKind.CollectionInitializerExpression ||
-                currentToken.Parent.Kind() == SyntaxKind.WithInitializerExpression))
+            // is { - property pattern clauses
+            if (currentToken.IsKind(SyntaxKind.OpenBraceToken) &&
+                (currentToken.Parent.IsKind(SyntaxKind.ObjectInitializerExpression) ||
+                currentToken.Parent.IsKind(SyntaxKind.CollectionInitializerExpression) ||
+                currentToken.Parent.IsKind(SyntaxKind.WithInitializerExpression) ||
+                currentToken.Parent.IsKind(SyntaxKind.PropertyPatternClause)))
```

Comment:
💡 The `IsKind` overload in namespace Microsoft.CodeAnalysis.CSharp.Extensions can simplify this. This can be done in a follow-up if it's not part of this PR.

```suggestion
                currentToken.Parent.IsKind(SyntaxKind.ObjectInitializerExpression, SyntaxKind.CollectionInitializerExpression, SyntaxKind.WithInitializerExpression, SyntaxKind.PropertyPatternClause))
```

Reply:
I prefer this to be a separate PR, as I also noticed a lot of other inconsistencies in this file (`.Kind == x` vs `.IsKind(x)`). Maybe a follow-up PR can iron out the usage inconsistencies and also use this simpler overload.